korio/src/commonMain/kotlin/com/soywiz/korio/net/ws/RawSocketWebSocketClient.kt

```diff
@@ -97,9 +97,11 @@ class RawSocketWebSocketClient(
             add("Sec-WebSocket-Version: 13")
             add("Connection: Upgrade")
             add("Sec-WebSocket-Key: ${key.toByteArray().toBase64()}")
-            add("Origin: $origin")
+            if (origin != null) {
+                add("Origin: $origin")
+            }
             add("User-Agent: ${HttpClient.DEFAULT_USER_AGENT}")
-        }.joinToString("\r\n") + "\r\n\n").toByteArray())
+        }.joinToString("\r\n") + "\r\n\r\n").toByteArray())
```

Comment:
This was the line that was causing the issue. The server expected `\r\n\r\n` and due to a typo missing the `\r` the websocket server thought the request was not completed